### DISCUSSION TITLE
BluRay-Content (m2ts) gets transcoded unneccessarily

### DISCUSSION THREAD

**Discussion by sic6SaNdMaN** [jellyfin/jellyfin#9678] [General] [answered]
Hi.

I have a ISO of BluRay movie.
When I copy over the folders "BDMV" and "CERTIFICATE" to my Jellyfin repo, the contained Video is handled different than when I throw the whole ISO into the repo.

Differences:

- File recognition:
-- Folders:
![image](https://user-images.githubusercontent.com/31123900/233980609-f4b1ae44-f0d1-408c-bb9f-5c60c7ce4f38.png)
-- ISO:
![image](https://user-images.githubusercontent.com/31123900/233980709-62cf1e2c-e13b-45b6-bf67-521c2ebf5efc.png)

- Playback:
-- Folders:
![Bildschirmfoto vom 2023-04-24 12-27-28](https://user-images.githubusercontent.com/31123900/233980487-e3ddb5d7-91c6-43e2-aafb-af9358225425.png)
-- ISO:
![Bildschirmfoto vom 2023-04-24 12-28-13](https://user-images.githubusercontent.com/31123900/233980518-72d0c9fc-df59-4bc3-a59d-3bf61c83eef7.png)
Which would be good, but I can't reach the Main Menu or the Movie when playing the ISO... but that's a separate issue...

My Setup:
- Sony Bravia Android TV with Kodi.
- Jellyfin client inside Kodi with "addon paths"
- Jellyfin server on Raspberry Pi 4 on raspbian, native install

When I play the m2ts file directly from the samba share with Kodi it just works fine - no need to transcode whatsoever - why should it?

So... why does Jellyfin transcode a simple 1080p SDR BluRay with stereo sound, when my setup is well capable of playing it?
Logs attached.

Thanks for any hints.

And here come the logs:
[jellyfin.log](https://github.com/jellyfin/jellyfin/files/11309939/jellyfin.log)
[FFmpeg.DirectStream-2023-04-24_12-26-44_098131fa0a0f08fae53bf05943c084a2_30343c65.log](https://github.com/jellyfin/jellyfin/files/11309960/FFmpeg.DirectStream-2023-04-24_12-26-44_098131fa0a0f08fae53bf05943c084a2_30343c65.log)

**Comment by Shadowghost** [comment, marked as answer]
This is ususally caused by ffmpeg handling ISOs way different and most of the time worse than it does the folders (in this sepcific case it seems to handle it better, it's unrealible nonetheless).
E.g. pure BDs usually have AVC video and not h264, additionally the language of the track from the ISO is not recognized - both is likely wrong for the ISO but correct for the folder. It's a limitation of ffmpeg/ffprobe.
https://github.com/jellyfin/jellyfin/pull/9254 tries fixing a lot of problems with our current non-ISO playback implementation. In 10.9 the recommended way of using full discs should be decrypted folders instead of ISOs.

**Reply by sic6SaNdMaN** [reply]
Thanks for the super-fast reply.

Well, OK, so it's a known issue.
Then I'll stay to the folder versions of BluRays, not ISOs, and wait for the version 10.9 :)

**Reply by sic6SaNdMaN** [reply]
Hmmm... but to clarify:
I don't have a problem copying over the folders.
My problem is, that the content is getting transcoded unneccessarily.
The recognition was correct for the folder version - AVC video and languages of the audio tracks - everything correct.

But than the video gets transcoded at playback time - which my raspberry can't handle and also shouldn't have to.

From the logs:
`[2023-04-24 12:26:43.806 +02:00] [INF] User policy for "Manu". EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: False EnableAudioPlaybackTranscoding: True
[2023-04-24 12:26:43.812 +02:00] [INF] StreamBuilder.BuildVideoItem( Profile="Kodi", Path="smb://xxx/Filme/yyy", AudioStreamIndex=null, SubtitleStreamIndex=null ) => ( PlayMethod=Transcode, TranscodeReason=VideoCodecNotSupported ) "media:/videos/098131fa-0a0f-08fa-e53b-f05943c084a2/stream.m3u8?MediaSourceId=098131fa0a0f08fae53bf05943c084a2&VideoCodec=h264,hevc,h265,mpeg4,mpeg2video,vc1&AudioCodec=aac,mp3,ac3,opus,flac,vorbis&AudioStreamIndex=1&VideoBitrate=999616000&AudioBitrate=384000&MaxFramerate=23.976025&api_key=<token>&TranscodingMaxAudioChannels=6&RequireAvc=false&Tag=543b6ca4c9f21c87d81daf7a932499c0&TranscodeReasons=VideoCodecNotSupported"`

So... my user policy forbids video transcoding, but it gets transcoded anyway and the reason is "VideoCodecNotSupported".
Why is that? Where does the information come from, that it isn't supported?

**Reply by Shadowghost** [reply]
The information comes from the client. The client announces its capabilities and the server reacts based on that. As you can see, avc is not in the supported video codec list: `VideoCodec=h264,hevc,h265,mpeg4,mpeg2video,vc1` this seems to be more of a bug in the Kodi extension to me (do note though that in the upcoming changes DVD and BD content is always remuxed due to it consisting of multiple files but the player requiring a single data stream).

**Reply by sic6SaNdMaN** [reply]
Again, thanks for the fast reply.

OK, yes, avc is definitively missing in the supported video codec list. Perhaps you could kindly hint the guys from the Kodi extension to this?

In the meanwhile I looked around a bit and saw, that I used "addon paths" (corrected my post).
This was just a performance test, if my pi could handle this also instead of native paths (yes, it can, more load than smb (nfs < smb < jellyfin), but not a problem at all).

Now I'm back to native paths.
Now it doesn't matter if I use the ISO oder folder version of the movie.
Kodi asks me wheter to play just the main movie or show the BluRay menu and plays the movie just fine - either way.

So... I totally forgot about switching to addon paths, I should have recognized this, because with native paths, jellyfin isn't even asked for the video content and transcoding isn't possible at all (the way I want it).

Well... so the problem is kinda solved for me, but I think there are some flaws in this whole issue that should be looked into nonetheless.

So, thanks for your support and somewhat pointing me to the right direction for a solution (for me).

**Reply by Shadowghost** [reply]
Since h264 and AVC are equivalent the proper fix would actually be to normalize both to h264 within the server, we're currently talking about how to approach this.

Sidenote: Native Paths might work but it's a hacky workaround because it essentially bypasses any restriction set by Jellyfin except metadata access.

**Reply by sic6SaNdMaN** [reply]
OK, yes, this sounds like a (simple?) solutiuon.
Perhaps you can clarify it in the GUIs, something like:
"Video: "1080p H264 SDR (normalized from AVC)"
or the likes.

Actually... this is all I ever wanted.
I just want to sync the media library, the "watched" or "continue" status all over my devices.
Nearly all of them have direct access to the server and samba is configured correctly.
And for my phone and the likes I can use the "addon path"-method or even JellyCon.
But I hardly even use that as I don't stream when I'm on the go.

So yeah, metadata access is exactly what I want and need :)

**Reply by Shadowghost** [reply]
Wouldn't make a service like trakt make more sense in that case? Jellyfin just for syncing watched state is a bit overkill imho :)

**Reply by sic6SaNdMaN** [reply]
Perhaps ;)
But first of all: I want to host everything myself.
And second: I like the interface, the possibility to organize the movies, make custom collections and everything. That are also very cool features that I use daily.
And having the possibility to remotely stream everything and/or download (yes, there are other ways to do that of course) is a nice addition, even if used only on very rare occasions.

I'm very happy with jellyfin and perhaps some suggestions will make it even better.
